test/FunctionsV2/TestCustomConnectionsStringResolver.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.Azure.WebJobs.Extensions.DurableTask;
+
+namespace WebJobs.Extensions.DurableTask.Tests.V2
```

Comment:
How does this differ from our default connection string resolver injected by our tests?

Reply:
The `TestConnectionStringResolver` always returns `TestHelpers.GetStorageConnectionString()` regardless of the `connectionStringName` that's passed in. This new one stores a dictionary of connection strings.

https://github.com/Azure/azure-functions-durable-extension/blob/cc7f8b20844a07b0729a3ae66bcc5dcf156baaf3/test/Common/TestConnectionStringResolver.cs#L10

Reply:
Can we just replace TestConnectionStringResolver()'s logic with this implementation (with a fallback to TestHlperes.GetStorageConnectionString() if the app setting doesn't exist?

Reply:
For this scenario, the code path expects `null` when an app setting doesn't exist. It then throws an `InvalidOperationException` which fails the test. If the connection string resolver returned `TestHelpers.GetStorageConnectionString()` as a default then the test would still pass because it got a connection string, but not the right one (passed through app settings).